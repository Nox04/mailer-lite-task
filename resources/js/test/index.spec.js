import Vue from 'vue';
import Vuetify from 'vuetify';
import { mount } from '@vue/test-utils';
import CustomField from '../components/Subscribers/Fields.vue';
import { singleField, multipleFields, emptyPreviousValues } from './__mocks__/FieldMocks';

Vue.use(Vuetify);

describe('Tests', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should mount without crashing', () => {
    const wrapper = mount(CustomField, {
      vuetify,
      propsData: {
        fields: multipleFields,
        previousValues: emptyPreviousValues,
      }
    })
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render different types of inputs acording to the props', () => {
    const wrapper = mount(CustomField, {
      vuetify,
      propsData: {
        fields: multipleFields,
        previousValues: emptyPreviousValues,
      }
    })
    expect(wrapper.findAll('input[role="switch"]').length).toBe(1);
    expect(wrapper.findAll('input:not([readonly]):not([role="switch"])').length).toBe(2);

    // Select using readonly inputs for DatePickers
    expect(wrapper.findAll('input[readonly]').length).toBe(1);
  })

  it('should render an error message on invalid state', async () => {
    const wrapper = mount(CustomField, {
      vuetify,
      propsData: {
        fields: singleField,
        previousValues: {},
      }
    })

    wrapper.find('input').setValue('something not numeric')

    //Needs to wait 2 ticks because of Vuetify bug.
    await Vue.nextTick()
    await Vue.nextTick()

    expect(wrapper.find('.v-messages__message').text()).toBe('Invalid number.');
  })

  it('should update model on input', async () => {
    const wrapper = mount(CustomField, {
      vuetify,
      propsData: {
        fields: singleField,
        previousValues: {},
      }
    })

    wrapper.find('input').setValue(123)
    await Vue.nextTick()

    expect(wrapper.vm.previousValues).toStrictEqual({ "1": "123" });
  })

})
