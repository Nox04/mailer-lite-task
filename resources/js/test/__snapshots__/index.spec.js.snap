// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests should mount without crashing 1`] = `
<div>
  <div class="row">
    <div class="col col-12">
      <div class="v-input theme--light v-text-field">
        <div class="v-input__control">
          <div class="v-input__slot">
            <div class="v-text-field__slot"><label for="input-3" class="v-label theme--light" style="left: 0px; position: absolute;">Count</label><input id="input-3" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <div class="v-messages__wrapper"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col col-12">
      <div class="v-menu">
        <div class="v-input v-input--is-readonly theme--light v-text-field">
          <div class="v-input__prepend-outer">
            <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate mdi mdi-calendar theme--light"></i></div>
          </div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-7" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input id="input-7" readonly="readonly" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <div class="v-messages__wrapper"></div>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col col-12">
      <div class="v-input theme--light v-text-field">
        <div class="v-input__control">
          <div class="v-input__slot">
            <div class="v-text-field__slot"><label for="input-11" class="v-label theme--light" style="left: 0px; position: absolute;">Name</label><input id="input-11" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <div class="v-messages__wrapper"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col col-12">
      <div class="v-input theme--light v-input--selection-controls v-input--switch">
        <div class="v-input__control">
          <div class="v-input__slot">
            <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-14" role="switch" type="checkbox" aria-disabled="false" value="">
              <div class="v-input--selection-controls__ripple"></div>
              <div class="v-input--switch__track theme--light"></div>
              <div class="v-input--switch__thumb theme--light">
                <!---->
              </div>
            </div><label for="input-14" class="v-label theme--light" style="left: 0px; position: relative;">Active</label>
          </div>
          <div class="v-messages theme--light">
            <div class="v-messages__wrapper"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
